lib/module.js

```diff
@@ -142,21 +139,25 @@ var warned = false;
 Module._findPath = function(request, paths) {
   if (path.isAbsolute(request)) {
     paths = [''];
+  } else if (!paths || paths.length === 0) {
+    return false;
   }
 
-  var cacheKey = JSON.stringify({request: request, paths: paths});
+  const cacheKey = JSON.stringify({request: request, paths: paths});
   if (Module._pathCache[cacheKey]) {
     return Module._pathCache[cacheKey];
   }
 
-  const exts = Object.keys(Module._extensions);
-  const trailingSlash = request.slice(-1) === '/';
+  var exts;
+  const trailingSlash = request.length > 0 &&
+                        request.charCodeAt(request.length - 1) === 47/*/*/;
 
   // For each path
-  for (var i = 0, PL = paths.length; i < PL; i++) {
+  for (var i = 0; i < paths.length; i++) {
```

Comment:
let apparently still has some optimization issues in V8. For hot code paths var is still preferred